controller/customer/index.js

```diff
@@ -4,4 +4,129 @@
+const CustomerModel = require("../../model/customer");
+
+const loginPassport = (req, res, next) => {
+  return new Promise((resolve, reject) => {
+    passport.authenticate("local", (err, customer, info) => {
+      if (err) return reject(err);
+      if (!customer) {
+        return resolve(
+          res.status(401).send(Utility.formatResponse(401, info.message))
+        );
+      }
+      req.logIn(customer, (err) => {
+        if (err) return reject(err);
+        const token = Utility.getSignedToken(customer.id);
+        resolve(res.status(200).send(Utility.formatResponse(200, { token })));
+      });
+    })(req, res, next);
+  });
+};
+
+const CustomerController = {
+  register: (req, res) => {
+    const payload = req.body;
+    return new Promise((resolve, reject) => {
+      Utility.createHash(payload.password)
+        .then((hash) => {
+          payload.password = hash;
+          CustomerModel.create({ ...payload })
+            .then((customer) => {
+              const token = Utility.getSignedToken(customer.id);
+              resolve(
+                res
+                  .status(200)
+                  .send(Utility.formatResponse(200, { token, id: customer.id }))
+              );
+            })
+            .catch((err) => {
+              reject(res.status(500).send(Utility.formatResponse(500, err)));
+            });
+        })
+        .catch((err) => {
+          reject(res.status(500).send(Utility.formatResponse(500, err)));
+        });
+    });
+  },
+
+  updateCustomer: (req, res) => {
+    const payload = req.body;
+
+    return new Promise((resolve, reject) => {
+      let updatePromise = Promise.resolve(payload);
+      if (payload.password) {
+        updatePromise = Utility.createHash(payload.password)
+          .then((hash) => {
+            payload.password = hash;
+            return payload;
+          })
+          .catch((err) => {
+            reject(
+              res.status(500).send(Utility.formatResponse(500, err.message))
+            );
+          });
+      }
+
+      updatePromise
+        .then((updatedPayload) => {
+          CustomerModel.update(
+            { ...updatedPayload },
+            { where: { id: payload.id } }
+          )
+            .then(() => {
+              resolve(
+                res
+                  .status(200)
+                  .send(Utility.formatResponse(200, `Updated Successfully`))
+              );
+            })
+            .catch((err) => {
+              reject(res.status(500).send(Utility.formatResponse(500, err)));
+            });
+        })
+        .catch((err) => {
+          console.error("Error updating customer: " + err);
+          reject(
+            res.status(500).send(Utility.formatResponse(500, err.message))
+          );
+        });
+    });
+  },
+
+  getCustomer: (req, res) => {
+    return new Promise((resolve, reject) => {
+      CustomerModel.findAndCountAll({ limit: 10, offset: 2 })
```

Comment:
limit and offset will come in the req, destructure from request eg: 
const {limit, offset} = req.body;